pymfy/api/devices/thermostat.py

```diff
@@ -29,11 +40,19 @@ def get_target_mode(self) -> str:
     def get_target_temperature(self) -> int:
         return cast(int, self.get_state("target_temperature"))
 
-    def get_target_end_date(self) -> datetime:
-        return datetime.utcfromtimestamp(cast(int, self.get_state("target_end_date")))
+    def get_target_end_date(self) -> Optional[datetime]:
+        timestamp = self.get_state("target_end_date")
+        if timestamp == -1:
+            return None
+        else:
+            return datetime.utcfromtimestamp(cast(int, timestamp))
 
-    def get_target_start_date(self) -> datetime:
-        return datetime.utcfromtimestamp(cast(int, self.get_state("target_start_date")))
+    def get_target_start_date(self) -> Optional[datetime]:
+        timestamp = self.get_state("target_start_date")
+        if timestamp == -1:
```

Comment:
@vlebourl Can this use case happened?

Reply:
I don't think so. As I understand it, end and start dates are epoch unix time stamps. In case of further_notice, the end date is set 60 seconds after the start date.